conjur/api/client.py

```diff
@@ -130,20 +135,28 @@ def __init__(self,
                             **loaded_config)
             self._api.login(login_id, password)
         else:
-            try:
-                credentials = CredentialsFromFile(DEFAULT_NETRC_FILE)
-                loaded_netrc = credentials.load(loaded_config['url'])
-            except netrc.NetrcParseError as netrc_error:
-                raise Exception("Error: netrc is in an invalid format. "
-                                f"Reason: {netrc_error}") from netrc_error
-            except Exception as exception:
-                # pylint: disable=line-too-long
-                raise RuntimeError("Unable to authenticate with Conjur. Please log in and try again.") from exception
+            if keyring.get_keyring().name in SUPPORTED_BACKENDS:
```

Comment:
This is a great question. If I am understanding correctly you are proposing that we allow the user to provide us with their own way to extract credentials from their keystore? Because we are using a third party library, we are confined to the keyring backends that they support. The keyring library gives us the ability to store creds in less secure ways (see list **here**) but not sure we want to support those as well since we have the netrc.

@InbalZilberman @sharonr78 WDYT? Should we also be supporting the less secure options and provide this flag for them to overwrite the keyring automatically chosen and even force the overwriting to the netrc.

Reply:
Just to be clear I'm not suggesting that the user could use the less secure Netrc. but instead use the keyring if it's possible.. like if the user use Rhel as it OS, and the user installed the needed dependencies to use our keyring.

Reply:
@sigalsax I understand from @mbenita-Cyberark that this is not relevant any more